src/encoded/auth0.py

```diff
@@ -96,6 +93,40 @@ def forget(self, request):
         return []
 
 
+def signup(request):
+    """
+    Create new user.
+
+    :param request: Pyramid request object
+    """
+    domain = 'encode.auth0.com'
+    url = "https://{domain}/userinfo?access_token={access_token}".format(domain=domain, access_token=request.json['accessToken'])
+    user_data_request = requests.get(url)
+
+    if user_data_request.status_code != 200:
+        raise HTTPBadRequest(explanation='Could not get user data')
+
+    users = request.registry[COLLECTIONS]['user']
+    user_data = user_data_request.json()
+
+    if not user_data['email_verified']:
+        raise HTTPBadRequest(explanation='Unverified e-mail')
+
+    user_info = {
+        'email': user_data['email'],
+        'first_name': user_data.get('given_name', '') or user_data.get('first_name', '') or user_data.get('name', ''),
+        'last_name': user_data.get('family_name', '') or user_data.get('last_name', ''),
+        'groups': [],
+        'submits_for': [],
+        'viewing_groups': [],
+        'status': 'current',
+    }
+    user = create_item(users.type_info, request, user_info)
+    if not user:
+        raise HTTPServerError(explanation='user not created')
```

Comment:
Is there a test for this?

Reply:
Technically yes; but I may need a more explicit one since you need to ask.

Reply:
Good pointed. I added unit test.